dvc/repo/plot.py

```diff
@@ -0,0 +1,208 @@
+        if default_plot:
+            data = []
+            for index, row in enumerate(csv.reader(fobj, delimiter=delimiter)):
+                assert len(row) >= 1
+                if index == 0 and len(row) > 1:
+                    # skip header
+                    continue
+                data.append({"y": row[-1], "x": index})
+        else:
+            data = [
+                row
+                for row in (
+                    csv.DictReader(
+                        fobj, skipinitialspace=True, delimiter=delimiter
+                    )
+                )
+            ]
+    return data
+
+
+def _load_from_tree(tree, datafile, default_plot=False):
+    filename = datafile.lower()
+    if filename.endswith(".json"):
+        data = _parse_json(datafile, default_plot, tree)
+    elif filename.endswith(".csv"):
+        data = _parse_csv(datafile, default_plot, tree)
+    elif filename.endswith(".tsv"):
+        data = _parse_csv(datafile, default_plot, tree, "\t")
+    elif filename.endswith(".yaml"):
+        data = _parse_yaml(datafile, default_plot, tree)
+    else:
+        raise PlotMetricTypeError(datafile)
+
+    return data
+
+
+def _load_from_revision(repo, datafile, revision, default_plot=False):
+    if revision is WORKSPACE_REVISION_NAME:
+        tree = repo.tree
+    else:
+        tree = repo.scm.get_tree(revision)
+
+    try:
+        data = _load_from_tree(tree, datafile, default_plot)
+        for d in data:
+            d["rev"] = revision
+    except FileNotFoundError:
+        raise NoMetricOnRevisionError(datafile, revision)
+    return data
+
+
+def _load_from_revisions(repo, datafile, revisions, default_plot=False):
+    data = []
+    exceptions = []
+
+    if len(revisions) <= 1:
+        if len(revisions) == 0 and repo.scm.is_dirty():
+            revisions.append("HEAD")
+        revisions.append(WORKSPACE_REVISION_NAME)
+
+    for rev in revisions:
+        try:
+            data.extend(
+                _load_from_revision(
+                    repo, datafile, rev, default_plot=default_plot
+                )
+            )
+        except NoMetricOnRevisionError as e:
+            exceptions.append(e)
+
+    if not data and exceptions:
+        raise NoMetricInHistoryError(datafile, revisions)
+    elif exceptions:
+        for e in exceptions:
+            logger.warning(
+                "File '{}' was not found at: '{}'. It will not be "
+                "plotted.".format(e.path, e.revision)
+            )
+    return data
+
+
+def _evaluate_templatepath(repo, template=None):
+    if not template:
+        return repo.plot_templates.default_template
+
+    if os.path.exists(template):
+        return template
+    return repo.plot_templates.get_template(template)
+
+
+@locked
+def plot(repo, datafile=None, template=None, revisions=None, file=None):
+    if revisions is None:
+        revisions = []
+
+    if not datafile and not template:
+        raise NoDataNorTemplateProvided()
```

Comment:
Hm, not a native speaker, but using `Nor` in exc is a bad idea. E.g. `no such file or directory` (no nor), so probably better `nor` -> `or` here too.